app/src/main/java/org/mozilla/fenix/settings/biometric/BiometricPromptPreferenceFragment.kt

```diff
@@ -0,0 +1,123 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.settings.biometric
+
+import android.app.Activity.RESULT_OK
+import android.app.KeyguardManager
+import android.content.Context
+import android.content.Intent
+import android.view.View
+import androidx.core.content.getSystemService
+import androidx.preference.Preference
+import androidx.preference.PreferenceFragmentCompat
+import mozilla.components.support.base.feature.ViewBoundFeatureWrapper
+import org.mozilla.fenix.ext.settings
+import org.mozilla.fenix.settings.requirePreference
+
+/**
+ * Helper for creating and implementing the [BiometricPromptFeature]. Currently used
+ * for logins and credit cards.
+ */
+abstract class BiometricPromptPreferenceFragment : PreferenceFragmentCompat() {
```

Comment:
Happy to see this, also really happy to see we're able to reduce code duplications. I just wished we could've also reduced showPinDialogWarning and showPinVerification to just this file as well.

Reply:
I think this can come in a followup as well